### DISCUSSION TITLE
Additional TUI functional requests

### DISCUSSION THREAD

**Discussion by pjfarleyiii** [Textualize/rich#1195] [Ideas]
I'll refer you to the bullet project (link below) for inspiration, but it would be awfully nice to be able to set up tabbed layouts with various kinds of selections to be picked:  Bulleted / radio button choices (pick one only), check-boxed choices (check one or more, returns a list or dict of selections made), and vertically scrolling versions of both of those kinds of choice lists, with support for arrow keys and HOME/END/PGUP/PGDN keys to move among the choices when the number of choices is larger than the size of the layout.

I recently added a pull request to the bullet project adding Windows support, coloring for prompts, and additional navigation keys for the bullet, check-box, and scrolling input formats.  I haven't received any feedback from the bullet developer on that pull yet.

I will second (or is that third or more?) other requests for scrolling text boxes, and I will add a request for scroll layout navigation (backwards and forwards) to the wish list.  IOW, to be able to tab to a scrolling text layout or panel and to be able to move up and down the scrolled text.  Not modify, just browse up and down, but also be able to highlight and copy text from that layout or panel.

With tabbed layouts and a variety of input selection methods available, I believe that rich can be a complete replacement for using curses to create TUI applications.

Thanks for the exceptional work you have done so far.

Regards,

Peter

bullet: [https://github.com/bchao1/bullet](url)

**Comment by willmcgugan** [comment]
There is a TUI branch in progress which will have these features (and more). No promise on the time scale though.

**Reply by pjfarleyiii** [reply]
Thank you for the prompt response.  I do understand about not promising any timeline.  Release it when it's done, and not before.

I look forward to seeing it though.

Regards,

Peter

**Comment by vladdu** [comment]
Hi! I can't see a TUI branch, is there any work done on this area?

**Reply by deadnews** [reply]
Maybe https://github.com/Textualize/textual?

**Reply by vladdu** [reply]
That is an option, but since there was at least an idea of some lightweight UI in Rich, I thought I'd ask.
